autoload/go/lint.vim

```diff
@@ -18,17 +14,23 @@ if !exists("g:go_errcheck_bin")
     let g:go_errcheck_bin = "errcheck"
 endif
 
-function! go#lint#Gometa(path_to_lint) abort
+function! go#lint#Gometa(...) abort
+    if a:0 == 0
+        let goargs = expand('%:p:h')
```

Comment:
> None of the other linters do recursive linting by default, why is GoMetaLinter a special case? There's no option to tell gometalinter to output full paths, so that's not a possible solution.

We know the current directory via Vim. So we can prepend it to gometalinter output if we want, but I think this would make it more complex than simpler.

> I understand not wishing to introduce breaking behaviour, but this is pretty fresh code, so there are probably not too many people affected, and at least this behaviour is consistent with the other linters.

Thinking now I think this is fair. Just let me use it a little bit and get a feeling for this.